README.rst

```diff
@@ -1198,6 +1245,181 @@ Once you have performed the 3 steps above, you can test it as follows:
+each containing two elements. The first element of the tuple should contain an
+identifier for the command and the second element should contain a ``dict``
+defining configuration of the command.
+
+Command Configuration
+^^^^^^^^^^^^^^^^^^^^^
+
+The ``dict`` defining configuration for command should contain following keys:
+
+1. ``label``
+""""""""""""
+
+A ``str`` defining label for the command used internally by Django.
+
+2. ``schema``
+"""""""""""""
+
+A ``dict`` defining `JSONSchema <https://json-schema.org/>`_ for inputs of command.
+You can specify the inputs for your command, add rules for performing validation
+and make inputs required or optional.
+
+Here is a detailed explanation of the schema used in above example:
+
+.. code-block:: python
+
+    {
+        # Name of the command displayed in "Send Command" widget
+        'title': 'Ping',
+        # Use type "object" if the command needs to accept inputs
+        # Use type "null" if the command does not accepts any input
+        'type': 'object',
+        # Specify list of inputs that are required
+        'required': ['destination_address'],
+        # Define the inputs for the commands along with their properties
+        'properties': {
+            'destination_address': {
+                # type of the input value
+                'type': 'string',
+                # label used for displaying this input field
+                'title': 'Destination Address',
+            },
+            'interface_name': {
+                'type': 'string',
+                'title': 'Interface Name',
+            },
+        },
+        # Error message to be shown if validation fails
+        'message': 'Destination Address cannot be empty'),
+        # Whether specifying addtionaly inputs is allowed from the input form
+        'additionalProperties': False,
+    }
+
+This example uses only handful of properties available in JSONSchema. You can
+experiment with other properties of JSONSchema for schema of your command.
+
+3. ``callable``
+"""""""""""""""
+
+A ``callable`` or ``str`` defining dotted path to a callable. It should return
+the command (``str``) to be executed on the device. Inputs of the command are
+passed as arguments to this callable.
+
+A callable for ``ping`` command in this example could like following:
+
+.. code-block:: python
+
+    def ping_command_callable(destination_address, interface_name=None):
+        command = f'ping -c 4 {destination_address}'
+        if interface_name:
+            command += f' -I {interface_name}'
+        return command
+
+Registering / Unregistering Commands
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+OpenWISP Controller provides registering and unregistering commands
+through utility functions ``openwisp_controller.connection.commands.register_command``
+and ``openwisp_notifications.types.unregister_notification_type``.
+Using these functions you can register or unregister commands from your code.
```

Comment:
We may specify that these functions are to be used as an alternative to the setting when developing custom modules based on openwisp-controller.